### ISSUE TITLE
Bee API stops to return data on GET requests (again!)

### ISSUE THREAD

**Issue by tmm360** [ethersphere/bee#2943]
### Context
Using Bee 1.5.1 with Docker, with addition of patch from #2909

### Summary
Very similar to #2909, node looks like blocked on reply to GET requests. Tried with a wget, it stays hanged.
@mrekucci @acud

I'm attaching logs from abort signal.
[abortLogs.txt](https://github.com/ethersphere/bee/files/8656157/abortLogs.txt)

### Expected behavior
It should reply with contents.

### Actual behavior
It doesn't reply, keeping connection open.

### Steps to reproduce
Don't know

### Possible solution
Don't know

**Comment by acud** [comment]
> Using Bee 1.5.1 with Docker, with addition of patch from #2909

Are you sure? the trace you've attached shows this line:
```
goroutine 23502523 [select, 13 minutes]:
github.com/ethersphere/bee/pkg/sharky.(*Store).Read(0xc00003e240, {0x187da00, 0xc00e34ce70}, {0x0, 0x0, 0x0}, {0xc008d4d950, 0x0, 0x0})
	github.com/ethersphere/bee/pkg/sharky/store.go:128 +0x205
```
Which should not be the case because if you've cherry picked the commit (using `git cherry-pick`) this now points to a comment which is not possible. Can you please attach your current `sharky/store.go` file?

**Comment by mrekucci** [comment]
In the bee repository do the following:
```
git fetch --tags
git checkout tags/v1.5.1 -b v1.5.1
git cherry-pick 6ce1d595
make binary
```
You'll find the built `bee` binary under the `./dist/` directory.

**Comment by tmm360** [comment]
I'm pretty sure that I did it right. I cherry picked `46ddd5c` and `a7751db` over tag `v1.5.1`, and I compiled with `docker build .`.
I still have sources, so I'm able to attach `sharky/store.go`: [store.zip](https://github.com/ethersphere/bee/files/8666623/store.zip)

I'm not at desk today, I can try to build again tomorrow.
Anyway, if it is useful, I also published the container that I'm currently using here https://hub.docker.com/r/etherna/bee-fix-2909
